src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java

```diff
@@ -383,7 +396,15 @@ private HttpResponse sendAndWait(HttpRequest request, HttpRequestOptions options
         }
         response = future.get();
       }
-      return httpResponseCreator.create(response, response.getResponseBodyAsStream());
+      HttpResponse httpResponse = httpResponseCreator.create(response, response.getResponseBodyAsStream());
+      if (httpResponse.getStatusCode() == 301 && httpResponse.getHeaders().containsKey("location")
+          && options.isFollowsRedirect()) {
+        if (currentRedirects > MAX_REDIRECTS) {
+          throw new IOException("Max redirects exceeded");
```

Comment:
Is there a reason why an IOException is used instead of a MaxRedirectException here?

Reply:
The method is defined in api and it should throw only IOException. I'm going to wrap the MaxRedirectException inside the IO.